app/models/network/conserving_producer.rb

```diff
@@ -5,18 +5,27 @@ class ConservingProducer < Technology
     extend Disableable
 
     def self.disabled?(options)
-      ! options[:flexibility]
+      options[:capping_solar_pv] == 'false'
     end
 
     def self.disabled_class
       Technology
     end
 
-    def conditional_consumption_at(frame)
-      production_at(frame)
+    def production_at(frame)
+      production = super
+      capping = (capacity * capping_percentage).abs
+
+      production > capping ? capping : production
     end
 
     def store(_frame, _amount)
     end
+
+    private
```

Comment:
No, not mandatory at all. It's a personal habit, and probably one I should stop. :laughing: